sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -302,42 +311,71 @@ class CompactBlockProcessor internal constructor(
                         } else {
                             "Done processing blocks"
                         }
-                        consecutiveChainErrors.set(0)
+
+                        resetErrorCounters()
+
                         val napTime = calculatePollInterval()
-                        Twig.debug {
+                        Twig.info {
                             "$summary Sleeping for ${napTime}ms " +
                                 "(latest height: ${_processorInfo.value.networkBlockHeight})."
                         }
                         delay(napTime)
                     }
+                    is BlockProcessingResult.ContinuityError -> {
+                        Twig.error {
+                            "Failed while processing blocks at height: ${result.failedAtHeight} with continuity " +
+                                "error: ${result.error}"
+                        }
+                        // No nap time set to immediately continue with the following block synchronization attempt
+                    }
                     is BlockProcessingResult.SyncFailure -> {
                         Twig.error {
                             "Failed while processing blocks at height: ${result.failedAtHeight} with: " +
                                 "${result.error}"
                         }
-                        // TODO [#1222]: Enrich BlockProcessingResult.SyncFailure with root cause
-                        // TODO [#1222]: https://github.com/zcash/zcash-android-wallet-sdk/issues/1222
-                        checkErrorResult(result.failedAtHeight)
+                        val failed = checkErrorAndFail(result.failedAtHeight, result.error)
+                        if (!failed) {
+                            val napTime = calculatePollInterval(true)
+                            Twig.info {
+                                "Sleeping for ${napTime}ms (latest height: ${_processorInfo.value.networkBlockHeight})."
+                            }
+                            delay(napTime)
```

Comment:
Nit: this code occurs three times and could be factored into a helper method.

Reply:
Just added a link to the filed issue to the code base. Thanks.